rtos/TARGET_CORTEX/mbed_rtx_idle.cpp

```diff
@@ -97,9 +91,9 @@ static void default_idle_hook(void)
 {
     uint32_t elapsed_ticks = 0;
 
+    uint32_t ticks_to_sleep = osKernelSuspend();
     core_util_critical_section_enter();
-    uint32_t ticks_to_sleep = svcRtxKernelSuspend();
-    if (ticks_to_sleep) {
+    if (!osRtxInfo.kernel.pendSV && ticks_to_sleep) {
         os_timer->schedule_tick(ticks_to_sleep);
```

Comment:
SysTimer (os_timer) was serialized to a critical section, but no longer is with this change. You'll either need to update SysTimer to add the additional synchronization or wait for #6534 to get merged.